Archipelago.MultiClient.Net/Helpers/LocationCheckHelper.cs

```diff
@@ -209,6 +277,58 @@ public void ScoutLocationsAsync(Action<LocationInfoPacket> callback = null, para
             // Maintain backwards compatibility if createAsHint parameter is not specified.
             ScoutLocationsAsync(callback, false, ids);
         }
+#else
+        /// <summary>
+        ///     Ask the server for the items which are present in the provided location ids.
+        /// </summary>
+        /// <param name="createAsHint">
+        ///     If true, creates a free hint for these locations.
+        /// </param>
+        /// <param name="ids">
+        ///     The locations ids which are to be scouted.
+        /// </param>
+        /// <remarks>
+        ///     Repeated calls of this method before a LocationInfo packet is received will cause the stored
+        ///     callback to be overwritten with the most recent call. It is recommended you chain calls to this method
+        ///     within the callbacks themselves or call this only once.
+        /// </remarks>
+        /// <exception cref="T:Archipelago.MultiClient.Net.Exceptions.ArchipelagoSocketClosedException">
+        ///     The websocket connection is not alive.
+        /// </exception>
+        public async Task<LocationInfoPacket> ScoutLocationsAsync(bool createAsHint, params long[] ids)
+        {
+            locationInfoPacketCallbackTask = new TaskCompletionSource<LocationInfoPacket>();
+            awaitingLocationInfoPacket = true;
+
+            await socket.SendPacketAsync(new LocationScoutsPacket()
+            {
+                Locations = ids,
+                CreateAsHint = createAsHint
+            });
+
+            return await locationInfoPacketCallbackTask.Task;
+        }

```

Comment:
```suggestion
        public Task<LocationInfoPacket> ScoutLocationsAsync(bool createAsHint, params long[] ids)
        {
            locationInfoPacketCallbackTask = new TaskCompletionSource<LocationInfoPacket>();
            awaitingLocationInfoPacket = true;

            await socket.SendPacketAsync(new LocationScoutsPacket()
            {
                Locations = ids,
                CreateAsHint = createAsHint
            });

            return locationInfoPacketCallbackTask.Task;
        }

```

Same here, shouldn't we _not_ await the `TaskCompletionSource.Task` so that it can be awaited by the caller?